product_cost_usd/models/sale_order_line.py

```diff
@@ -0,0 +1,45 @@
+import pdb
+from odoo import api, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    @api.depends("product_id", "company_id", "currency_id", "product_uom")
+    def _compute_purchase_price(self):
+        """Inherited to recalculate purchase price when pricelist item is
+        based on cost in usd.
+        """
+        res = super()._compute_purchase_price()
+        for line in self:
+            date = line.order_id.date_order
+            price_rule = line.order_id.pricelist_id._compute_price_rule([(line.product_id, 1, False)])
+            price, rule = price_rule[line.product_id.id]
+            suitable_rule = line.order_id.pricelist_id.item_ids.filtered(lambda x: x.id == rule)
+            if not suitable_rule or suitable_rule.base != "standard_price_usd":
+                return res
+            frm_cur = self.env.ref("base.USD")
+            to_cur = line.order_id.pricelist_id.currency_id
+            purchase_price = line.product_id.standard_price_usd
+            if line.product_uom != line.product_id.uom_id:
+                purchase_price = line.product_id.uom_id._compute_price(purchase_price, line.product_uom)
+            line.purchase_price = frm_cur._convert(purchase_price, to_cur, self.env.company, date, round=False)
+            return res
```

Comment:
This is inside a loop, but you're returning, so only the first record is being considered